Client/Frontend/Browser/TabManager.swift

```diff
@@ -173,6 +173,11 @@ class TabManager: NSObject {
             return
         }
 
+        // Make sure to wipe the private tabs if the user has the pref turned on
+        if shouldClearPrivateTabs(), !(tab?.isPrivate ?? false) {
+            removeAllPrivateTabsAndNotify(false)
```

Comment:
Didnt notice that line. My bad. But I still think that we dont really need to check the previous tab to be sure when to delete the private tabs